### PR TITLE
Additional examples in doc of `tested-with` (fix #4894)

### LINKED ISSUE #4894: Documentation of field `tested-with`
The documentation http://cabal.readthedocs.io/en/latest/developing-packages.html?highlight=tested-with#pkg-field-tested-with currently just says
```
tested-with: compiler list

    A list of compilers and versions against which the package has been tested (or at least built).
```
The format of the `compiler list` needs to be explained by a representative example at least.

Cabal does have specific expectations here, as witnessed by this error:
```
cabal: <loc>: Parse of field 'tested-with' failed.
```

### PR DESCRIPTION
**Pull request by andreasabel**
Fix #4894: document alternatives to write a tested-with list
    
A grab-bag of alternatives as discussed at haskell/cabal#4894.

This PR is based on PR #7603 to get extra sphinx sanity checking.

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
Perfect.

File: doc/cabal-package.rst [hunk 1]
```diff
@@ -629,19 +629,78 @@ describe the package as a whole:
...
+          , GHC == 8.4.4
+          , GHC == 8.2.2
+          , GHC == 8.0.2
+          , GHC == 7.10.3
+          , GHC == 7.8.4
+          , GHC == 7.6.3
+          , GHC == 7.4.2
+
+
+    Second, a concise set notation syntax is available:
```
Comment (Mikolaj) [inline doc/cabal-package.rst:699]: Fourth? Additionally?

Comment (andreasabel) [reply doc/cabal-package.rst:699]: Ah, this was supposed to continue the "First" from features available starting with 3.0.  Maybe the context is lost over the long display.  I could repeat, like
```
Secondly, starting with 3.0, a concise set notation...
```
Better?

Comment (Mikolaj) [reply doc/cabal-package.rst:699]: Oh, no, the original is better. You just make bold assumptions about people's attention spans. :)

I guess a new paragraph for "First" would have helped me, but feel free to ignore.

Comment (andreasabel) [reply doc/cabal-package.rst]: > bold assumptions about people's attention spans. :)

Haha, typical mistake of a writer, this is why it is good to have proof readers!

I changed this to use an enumeration (clarifies structure).

**Review by emilypi — APPROVED** [review]

**Comment by emilypi** [comment]
Looks good @andreasabel! Please resolve the merge conflicts and i'm happy to merge into master :)

**Comment by Mikolaj** [comment]
@andreasabel: I hope the problems are gone now.

**Pull request merged**